internal/corazawaf/rulegroup.go

```diff
@@ -31,6 +31,17 @@ func (rg *RuleGroup) Add(rule *Rule) error {
 	if rg.FindByID(rule.ID_) != nil && rule.ID_ != 0 {
 		return fmt.Errorf("there is a another rule with id %d", rule.ID_)
 	}
+
+	numInferred := 0
+	rule.inferredPhases[rule.Phase_] = true
+	for _, v := range rule.variables {
+		min := minPhase(v.Variable)
+		if min != 0 {
+			rule.inferredPhases[min] = true
```

Comment:
```suggestion
			// multiphaseEvaluation logic: adds to inferredPhases the earliest phase a variable may be populated, therefore it may be evaluated. 
			rule.inferredPhases[min] = true
```